env/jun19/sim_no-threads.yaml

```diff
@@ -1,7 +1,7 @@
 spack:
   definitions:
     - when: platform == 'linux'
-      root: [root@6.16.00+fortran+gdml+http+memstat+pythia6+pythia8+vc~vdt~python~tmva]
+      root: [root@6.16.00+fortran+gdml+http+memstat+pythia6+pythia8+vc~vdt~python~tmva ^mesa~llvm ^python@2.7.16]
```

Comment:
Could you try moving the `^python@2.7.16` as a `- python@2.7:2.99` as its own entry in the `specs:` section to the `prce+jit` part to aid the concretizer?

Reply:
Okay, I think, you can leave out the `^python@2.7.16` bit completely, because the `depends_on` should take care of it. This works nicely on my local machine.

Reply:
Test shows that the concretizer fails, if one does not specify python@2.7.16 in the environment